### PR TITLE
Fix oneshot parameters to actually work

### PR DESCRIPTION
**Pull request by bertyah**
The dict of parameters was not working as intended. I fixed it to work correctly and also added the offset parameter.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Change Log
 
+## 2.2.1
```
Comment (arm4b) [inline CHANGES.md]: Let's bump the minor version here per https://semver.org/#spec-item-7 as the PR introduces the new action param.

```suggestion
## 2.3.0
```

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
+
+- Refactor params for Oneshot search
+- Change ResultReader to JSONResultReader
```
Comment (arm4b) [inline CHANGES.md]: Worth clarifying the Changelog what this change helped to solve/fix
```suggestion
- Change ResultReader to JSONResultReader to fix ...
```

File: pack.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@ keywords:
   - splunk
   - monitoring
   - alerting
-version: 2.2.0
+version: 2.2.1
```
Comment (arm4b) [inline pack.yaml]:
```suggestion
version: 2.3.0
```

File: CHANGES.md (hunk 2 above)
Comment (bertyah) [reply CHANGES.md]: updated the reason!

**Review by arm4b — APPROVED** [review]
Thanks for the contribution!

**Comment by bertyah** [comment]
Thanks for the quick review :)

**Pull request merged**